### PR TITLE
feat: adapt PlatformWiring to expect system transactions as a return parameter from TransactionPrehandler and StateHasher

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:
This PR adapts `PlatformWiring`, so that it consumes system transactions as return objects from TransactionPrehandler and StateHasher components which do preHandle and handleRound logic.

Their return values will be populated from the Consumer<> callbacks registered in SwirldState API in #16703 

**Related issue(s)**: #16707 

Fixes #

**Notes for reviewer**:
What is left here is to consume the system transactions from the callback registered as an argument in #16703

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java [hunk 1]
```diff
@@ -95,5 +100,8 @@ public void prehandleApplicationTransactions(@NonNull final PlatformEvent event)
 
             preHandleTime.update(startTime, time.nanoTime());
         }
+
+        // TODO temporarily return no transactions until SwirldState.preHandle() is updated with callback
+        return new ArrayList<>();
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java:105]: return Collection.EMPTY_LIST

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java [hunk 2]
```diff
@@ -133,10 +137,13 @@ public void handleConsensusRound(final ConsensusRound round) {
...
      * @param round the round to seal
      */
-    public void sealConsensusRound(@NonNull final Round round) {
+    public List<ScopedSystemTransaction<StateSignatureTransaction>> sealConsensusRound(@NonNull final Round round) {
         Objects.requireNonNull(round);
         final MerkleRoot state = stateRef.get();
         state.getSwirldState().sealConsensusRound(round);
+
+        // TODO temporary return no transactions until the SwirldState interface is updated to register callbacks
+        return new ArrayList<>();
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/SwirldStateManager.java:146]: Collection.EMPTY_LIST

**Comment by lpetrovic05** [comment]
I think there is no need to have 2 pathways for system transaction in the platform, we can switch over to the new pathway immediately. In the short term, the apps would not parse the system transactions, but the handler and prehandler would separate them out. This way, we can completely remove the old pathway and replace it with the new.
Also keep in mind that the ISS detector also needs to change where its getting its transactions from

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java [hunk 3]
```diff
@@ -102,5 +104,9 @@ public void prehandleApplicationTransactions(@NonNull final PlatformEvent event)
 
             preHandleTime.update(startTime, time.nanoTime());
         }
+
+        // TODO adapt this logic to read transactions directly from the callback passed in SwirldState.preHandle() when
```
Comment (lpetrovic05) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java:108]: did you check with services if the prehandle call is synchronous?

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java (hunk 3 above)
Comment (IvanKavaldzhiev) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java:108]: Services confirmed they execute preHandle in parallel, which can be seen [here](https://github.com/hashgraph/hedera-services/blob/93693c3adb38c668ebaeb02f3e86e9b06c89475a/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java#L141). Each preHandle call should have a separate synchronous execution and blocks until it fully executes.

I think we should be good.

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**